os-3.10-multus/scripts/provision.sh

```diff
@@ -0,0 +1,216 @@
+
+# Create ansible inventory file
+cat >$inventory_file <<EOF
+[OSEv3:children]
+masters
+nodes
+
+[OSEv3:vars]
+ansible_ssh_user=root
+ansible_ssh_pass=vagrant
+deployment_type=origin
+openshift_deployment_type=origin
+openshift_clock_enabled=true
+openshift_master_identity_providers=[{'name': 'allow_all_auth', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]
+openshift_disable_check=memory_availability,disk_availability,docker_storage,package_availability,docker_image_availability
+openshift_image_tag=v3.10.0-rc.0
+ansible_service_broker_registry_whitelist=['.*-apb$']
+openshift_hosted_etcd_storage_kind=nfs
+openshift_hosted_etcd_storage_nfs_options="*(rw,root_squash,sync,no_wdelay)"
+openshift_hosted_etcd_storage_nfs_directory=/opt/etcd-vol
+openshift_hosted_etcd_storage_volume_name=etcd-vol
+openshift_hosted_etcd_storage_access_modes=["ReadWriteOnce"]
+openshift_hosted_etcd_storage_volume_size=1G
+openshift_hosted_etcd_storage_labels={'storage': 'etcd'}
+openshift_node_kubelet_args={'max-pods': ['40'], 'pods-per-core': ['40']}
+openshift_master_admission_plugin_config={"ValidatingAdmissionWebhook":{"configuration":{"kind": "DefaultAdmissionConfig","apiVersion": "v1","disable": false}},"MutatingAdmissionWebhook":{"configuration":{"kind": "DefaultAdmissionConfig","apiVersion": "v1","disable": false}}}
+os_sdn_network_plugin_name='redhat/openshift-ovs-networkpolicy'
+
+[nfs]
+node01 openshift_ip=$master_ip
+
+[masters]
+node01 openshift_ip=$master_ip
+
+[etcd]
+node01 openshift_ip=$master_ip
+
+[nodes]
+node01 openshift_schedulable=true openshift_ip=$master_ip openshift_node_group_name="node-config-all-in-one"
+EOF
+
+# Add cri-o variable to inventory file
+if [[ $1 == "true" ]]; then
+    sed -i 's/\[OSEv3\:vars\]/\[OSEv3\:vars\]\nopenshift_use_crio=true\n/' $inventory_file
+fi
+
+# Install prerequisites
+ansible-playbook -e "ansible_user=root ansible_ssh_pass=vagrant" -i $inventory_file $openshift_ansible/playbooks/prerequisites.yml
+ansible-playbook -i $inventory_file $openshift_ansible/playbooks/deploy_cluster.yml
+
+# Create OpenShift user
+/usr/bin/oc create user admin
+/usr/bin/oc create identity allow_all_auth:admin
+/usr/bin/oc create useridentitymapping allow_all_auth:admin admin
+/usr/bin/oc adm policy add-cluster-role-to-user cluster-admin admin
+
+# Create local-volume directories
+for i in {1..10}
+do
+  mkdir -p /var/local/kubevirt-storage/local-volume/disk${i}
+  mkdir -p /mnt/local-storage/local/disk${i}
+  echo "/var/local/kubevirt-storage/local-volume/disk${i} /mnt/local-storage/local/disk${i} none defaults,bind 0 0" >> /etc/fstab
+done
+chmod -R 777 /var/local/kubevirt-storage/local-volume
+
+# Setup selinux permissions to local volume directories.
+chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
+# Add privileged to local volume provision service account
+/usr/bin/oc adm policy add-scc-to-user privileged -z local-storage-admin
+
+oc apply -f ./multus.yml
```

Comment:
I see you run apply but you do not really wait until it will download all needed images.

Reply:
I will add a while before exit the script to check that all the pods related to the daemonset are running. I hope this will fix the problem on the standard CI.